### ISSUE TITLE
p10k prompt is lost upon restart or new tab

### ISSUE THREAD

**Issue by justiceamoh** [warpdotdev/Warp#5092]
### Dupe Check

- [X] I have [searched Warp bugs](https://github.com/warpdotdev/warp/issues?q=is%3Aissue+label%3ABugs) and there are no duplicates

### Describe the bug

After I installed and configured powerlevel10k, Warp correctly displayed the prompt as desired. However, immediately after I open a new terminal or a new tab, it appears to use a different prompt configuration altogether. I've attached pictures illustrating the issue.

### To reproduce

1. Run `p10k configure`
2. Configure the prompt similar to what's shown above in the pictures: use Prompt Height 1
3. ✅ Warp should correctly render the configured prompt
4. Open a new tab, a new window or a new split screen
5. ❌ Warp now reverts to a different prompt configuration

### Expected behavior

I expect warp to retain the p10k prompt configuration across new terminal sessions.

### Screenshots

![CleanShot 2024-07-03 at 08 25 05@2x](https://github.com/warpdotdev/Warp/assets/2006900/92bdd549-7b5d-4896-bf02-2a517b0dbb54)

![CleanShot 2024-07-03 at 08 28 10@2x](https://github.com/warpdotdev/Warp/assets/2006900/c0aaa88f-189f-4b3c-9b51-874581665036)
 
![CleanShot 2024-07-03 at 08 29 10@2x](https://github.com/warpdotdev/Warp/assets/2006900/f242e820-da7f-480b-a4a6-845de92e2149)

### Operating system

MacOS

### Operating system and version

Sonoma 14.5

### Shell Version

zsh 5.9 (x86_64-apple-darwin23.0)

### Current Warp version

v0.2024.06.25.08.02.stable_01

### Regression

No, this bug or issue has existed throughout my experience using Warp

### Recent working Warp date

N/A

### Does this block you from using Warp daily?

Yes, this issue prevents me from using Warp daily.

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, this I confirmed this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b9d78064-c89e-4973-b153-5178a31ee54e

None

**Comment by dannyneira** [comment]
Hi @justiceamoh Sorry to hear about this. Does it help at all if you restart Warp after running p10k configure? 

I'm unable to reproduce this out of the box p10k. I suggest you check your `~/.zshrc` file for anything outside the normal source lines for p10k, there maybe another source line after this that's applying another prompt or oh-my-zsh theme.
```sh
  ## Powerlevel10k ; To customize prompt, run `p10k configure` or edit ~/.p10k.zsh
if [[ -r "${XDG_CACHE_HOME:-$HOME/.cache}/p10k-instant-prompt-${(%):-%n}.zsh" ]]; then
  source "${XDG_CACHE_HOME:-$HOME/.cache}/p10k-instant-prompt-${(%):-%n}.zsh"
fi
source ~/powerlevel10k/powerlevel10k.zsh-theme
```

**Comment by justiceamoh** [comment]
Thank you for getting back to me, @dannyneira. I found the solution to my issue, which was related to how I installed and applied the `p10k` theme. Initially, I installed it into the `oh-my-zsh` themes directory and applied the theme as a `ZSH_THEME`. However, I resolved the issue by reverting to manual installation of `p10k` and configuring it by running `exec zsh` right after the install (as directed in the [tutorial video](https://www.youtube.com/watch?v=dIV9Cso4Mi8)). Thanks a lot for your help!
